pkg/ring/kv/dynamodb/dynamodb.go

```diff
@@ -176,38 +183,51 @@ func (kv dynamodbKV) Batch(ctx context.Context, put map[dynamodbKey][]byte, dele
 		return nil
 	}
 
-	writeRequests := make([]*dynamodb.WriteRequest, 0, writeRequestSize)
+	writeRequestsSlices := make([][]*dynamodb.WriteRequest, int(math.Ceil(float64(writeRequestSize)/float64(DdbBatchSizeLimit))))
+	for i := 0; i < len(writeRequestsSlices); i++ {
+		writeRequestsSlices[i] = make([]*dynamodb.WriteRequest, 0, DdbBatchSizeLimit)
+	}
+
+	currIdx := 0
 	for key, data := range put {
 		item := kv.generatePutItemRequest(key, data)
-		writeRequests = append(writeRequests, &dynamodb.WriteRequest{
+		writeRequestsSlices[currIdx] = append(writeRequestsSlices[currIdx], &dynamodb.WriteRequest{
 			PutRequest: &dynamodb.PutRequest{
 				Item: item,
 			},
 		})
+		if len(writeRequestsSlices[currIdx]) == DdbBatchSizeLimit {
+			currIdx++
+		}
 	}
 
 	for _, key := range delete {
 		item := generateItemKey(key)
-		writeRequests = append(writeRequests, &dynamodb.WriteRequest{
+		writeRequestsSlices[currIdx] = append(writeRequestsSlices[currIdx], &dynamodb.WriteRequest{
 			DeleteRequest: &dynamodb.DeleteRequest{
 				Key: item,
 			},
 		})
+		if len(writeRequestsSlices[currIdx]) == DdbBatchSizeLimit {
+			currIdx++
+		}
 	}
 
-	input := &dynamodb.BatchWriteItemInput{
-		RequestItems: map[string][]*dynamodb.WriteRequest{
-			*kv.tableName: writeRequests,
-		},
-	}
+	for _, slice := range writeRequestsSlices {
+		input := &dynamodb.BatchWriteItemInput{
+			RequestItems: map[string][]*dynamodb.WriteRequest{
+				*kv.tableName: slice,
+			},
+		}
 
-	resp, err := kv.ddbClient.BatchWriteItemWithContext(ctx, input)
-	if err != nil {
-		return err
-	}
+		resp, err := kv.ddbClient.BatchWriteItemWithContext(ctx, input)
+		if err != nil {
+			return err
+		}
 
-	if resp.UnprocessedItems != nil && len(resp.UnprocessedItems) > 0 {
-		return fmt.Errorf("error processing batch request for %s requests", resp.UnprocessedItems)
+		if resp.UnprocessedItems != nil && len(resp.UnprocessedItems) > 0 {
+			return fmt.Errorf("error processing batch request for %s requests", resp.UnprocessedItems)
```

Comment:
Are we planning to log the unprocessed items using `%s` string?
From the error message itself I thought we are only going to log the number of requests.

Reply:
Yes, the idea is to give some output that might help investigate why some actions were not completed. you are more concern if this data is useful or just the message is not very clear?

Reply:
I thought you intend to use `%d` instead of `%s`. 
I am not sure what is the type of `UnprocessedItems`. Is it a []string or a slice of more complicated items? If it is strings I think that's fine, if it is other type then it might have some readability issue.

Reply:
Looks like this is what we have from your last pr. Then I think it is fine. We can improve in the future if this is indeed an problem.

Reply:
Ah got it. It is basically a subset of the request which failed. So yes, it is bit more complex than string. We can follow up in another PR